docs/pages/hdmi_cec_source_tests.md

```diff
@@ -0,0 +1,176 @@
+
+- Get Logical address
+- Get Physical address
+- Syncronous transmission, and communicating via hotplug connectivity
+
+## Test scenarios
+
+The HAL CEC layer facilitates the transmission and reception of CEC information on the CEC bus. It does not handle any specific opcode commands, nor does it validate supported HAL CEC opcodes for sending or receiving.
+
+It is the responsibility of the caller to manage the opcodes. The current test cases will verify responses from connected devices for a subset of opcodes as part of the testing process.
+
+|S.No.|Test Functionality|Description|
+|-----|------------------|-----------|
+| 1 |[Logical address](#logical-address-discovery)|Facilitating the Discovery of logical addresses getting, and removing the logical address of the device (for sink devices) |
+| 2| [Physical Address](#physical-address)| Retrieving the physical address |
+| 3| [CEC Synchronous Transmission](#cec-synchronous-transmission)| Transmitting CEC frames and acknowledging them |
+| 4| [CEC Receive functionality](#cec-receive-functionality)| Receiving CEC information from other devices and communicating it to the above layers through registered callback functions |
+| 5| [CEC HotPlug Functionality](#cec-hotplug-functionality)| Managing CEC during Hotplug and HotUnplug events |
+
+-----------
+
+## Logical Address Discovery
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 1 |[Logical address](#logical-address-discovery)|Get the logical address of the `DUT`. This will add the logical address, as per source functionality. |HdmiCecGetLogicalAddress|Y|NA
+
+### Emulator Requirements
+
+- Boot with control configuration with various configurations having a predefined set of nodes:
+  - configuration to support the discovery of logical addresses. The caller provides the logical address, and HAL checks the availability of this address and feedback the same to the caller. 
+  - Verify for the valid logical address and return the appropriate error code based on the logical address availability.  
+
+### Control Plane Requirements
+
+- The control plane will allow removing or adding a node to the network.
+  - allowing adding/removing node sink node logical address 0
+  - allowing adding/removing source node
+  - Support the CEC commands from the external devices on L3 Test Cases. 
+
+### Physical Address
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 1| [Physical Address](#physical-address)| Verify the valid physical address allocated through the HAL function.|HdmiCecGetPhysicalAddress| Y  |NA||
```

Comment:
Enable the television connected to the DUT first so..

Reply:
Updated as corrected